powerstore/resource_volume.go

```diff
@@ -416,7 +496,7 @@ func (r resourceVolume) Update(ctx context.Context, req tfsdk.UpdateResourceRequ
 		return
 	}
 
-	updateVolState(&state, volResponse, hostMapping, &plan, "Update")
+	updateVolState(&state, volResponse, hostMapping, volGroupMapping, &plan, "Update")
```

Comment:
same as above
try to use https://github.com/dell/terraform-provider-powerstore/blob/dev/powerstore/defaults.go#L14

also, try to make updateVolState as method instead of function